libs/genai/langchain_google_genai/_function_utils.py

```diff
@@ -47,29 +42,36 @@ class _ToolDict(TypedDict):
     function_declarations: Sequence[_FunctionDeclarationLike]
 
 
+GeminiToolLike = Union[
+    GoogleTool,
+    _ToolDict,
+    _FunctionDeclarationLike,
+]
+
+
 def convert_to_genai_function_declarations(
-    tool: Union[
-        GoogleTool,
-        _ToolDict,
-        _FunctionDeclarationLike,
-        Sequence[_FunctionDeclarationLike],
-    ],
+    tool: GeminiToolLike,
 ) -> GoogleTool:
     if isinstance(tool, GoogleTool):
         return cast(GoogleTool, tool)
-    if isinstance(tool, type) and issubclass(tool, BaseModel):
+    if isinstance(tool, (BaseTool, Type[BaseModel], dict, FunctionDeclaration)):
```

Comment:
I believe, we want to distinguish a case when dict is OpenAPI spec vs dict is like {"function_declarations": ...}.

Reply:
I added some comments and dict key checking.